packages/opds-web-client/src/components/Collection.tsx

```diff
@@ -210,13 +210,15 @@ export default class Collection extends React.Component<CollectionProps, {}> {
     return (
       !this.props.hidden &&
       !this.props.isFetchingPage &&
-      this.props.collection.nextPageUrl
+      !!this.props.collection.nextPageUrl
     );
   }
 
   fetch() {
-    if (this.canFetch()) {
-      this.props.fetchPage(this.props.collection.nextPageUrl);
+    // had to move this typeguard into here for typescript to
+    // believe the nextPageUrl was defined.
+    if (this.canFetch() && this.props.collection.nextPageUrl) {
```

Comment:
One annoying thing about typescript is that you can't save a typeguard into a variable:
```js
function(a: string | null) {
  const isString = !!string;
  if (isString){
    return a.toUppercase();
}
// you'll get an error at toUppercase above, because typescript doesn't realize the type has been narrowed by isString
if(a) {
  return a.toUppercase();
} 
// the typeguard has to be inline for it not to error
```

Reply:
@EdwinGuzman re your commend on the type guard I inlined that used to be in the `showFooter` var.